lib/services/local/scoped_tokens.go

```diff
@@ -92,7 +99,9 @@ func (s *ScopedTokenService) UseScopedToken(ctx context.Context, name string) (*
 		return nil, trace.Wrap(err)
 	}
 	if err := joining.ValidateTokenForUse(token); err != nil {
-		if trace.IsLimitExceeded(err) {
+		// unscoped tokens are automatically deleted on use after expiration,
+		// so we do the same here for parity
+		if errors.Is(err, joining.ErrTokenExpired) {
 			if err := s.svc.DeleteResource(ctx, name); err != nil {
 				return nil, trace.LimitExceeded("cleaning up expired token: %v", err)
 			}
```

Comment:
This is a race condition, the token might've changed between the Get and now; it's also unfair to punish the caller with a `LimitExceeded` due to a failed delete (which might've happened because of another concurrent `UseScopedToken`) if all they wanted was to confirm that a given token was usable.